app/assets/javascripts/index.js

```diff
@@ -412,6 +412,9 @@ $(document).ready(function () {
 
     if (OSM.router.route(this.pathname + this.search + this.hash)) {
       e.preventDefault();
+      if (this.pathname !== "/directions") {
```

Comment:
The code here closes the menu after clicking on any menu item except *directions*. By *menu* I mean everything that shows up when the burger button is clicked. The directions form is inside this menu, therefore an exception is made for it, the menu should stay open to keep the form visible.

app\assets\javascripts\index\directions.js:123:5 happens after the route endpoints are received, then you can hide the form and the rest of the menu.

Reply:
app\assets\javascripts\index\search.js:13:5 sets "prevent default" to the directions button `e.preventDefault()` and app\assets\javascripts\index.js:399:9 checks if the `<a>` tag is "prevent default" `e.isDefaultPrevented()`. Therefore, returns empty value and never reaches `this.pathname !== "/directions"` if statement.

Reply:
Why should it reach `this.pathname !== "/directions"`?

If you're at app\assets\javascripts\index\search.js:13:5, you have the menu open. And you want it to stay open because you're switching from the search form to the directions form. `if (this.pathname !== "/directions") ...` closes the menu, something you don't want at this point.